crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -110,6 +133,63 @@ impl C8yMapperActor {
         Ok(())
     }
 
+    /// Performs auto-registration process for an entity under a given
+    /// identifier.
+    ///
+    /// If an entity is a service, its device is also auto-registered if it's
+    /// not already registered.
+    ///
+    /// It returns MQTT register messages for the given entities to be published
+    /// by the mapper, so other components can also be aware of a new device
+    /// being registered.
```

Comment:
This is a good idea. But I would be cautious. Indeed, the measurement/alarm/event must not be sent *before* the device is actually registered on C8Y. But, the current code jumps into the translation.